### ISSUE TITLE
Piping async output into stdin breaks visidata keyboard input

### ISSUE THREAD

**Issue by ovikk13** [saulpw/visidata#1347]
**Small description**
Piping output from an async nodejs application into visidata results in visidata not accepting user input. This only occurs when output is sent to stdout before the application is done.

Instead of keybindings working correctly, the text is written at the top bar, as if visidata is a background image you are overwriting.

Tested in both bash and fish-shell

**Expected result**
Visidata accepts user input

**Actual result with screenshot**
![image](https://user-images.githubusercontent.com/16265319/162283336-ae1e7f12-4275-4535-825b-50acbbd0ce88.png)

**Steps to reproduce with sample data and a .vd**
A sample nodejs application `test.js`:
```js
console.log("test1");
setTimeout(() => {
   console.log("test2");
}, 1000);
```
At a terminal: `$ nodejs test.js | vd`

**Additional context**
Program | Version
----------- | ------------
VisiData |  v2.8
Python   | 3.10.4
Terminal | Alacrity version  0.10.1-1
OS         | Arch Linux

Thanks for the help! :)

**Comment by anjakefala** [comment]
Thanks so much for the repro!

**Comment by anjakefala** [comment]
Could be a big clue for: https://github.com/saulpw/visidata/issues/1319

**Comment by saulpw** [comment]
Notes:
- Only the immediate output must be present to cause the issue; the delayed output is not necessary.  The delay to process exit seems to be the culprit.
- This seems to be nodejs specific; `(echo foo; sleep 1; echo bar) | vd` works fine.

**Comment by saulpw** [comment]
It appears that the given nodejs repro case causes curses to exit raw mode: the tty attrs on stdin in vd (from `tcgetattr()`) are modified at nodejs exit, but only if it is delayed.  I don't know why/how this is happening, but always calling `curses.reset_prog_mode()` (since we call `def_prog_mode()` just after initializing curses) at the start of `vd.getkeystroke()` seems to fix it.  This may also fix #1319.
